panelapp/webservices/views.py

```diff
@@ -127,64 +127,68 @@ def get_panel(request, panel_name):
     if "version" in request.GET:
         version = request.GET["version"]
         try:
-            major_version = int(version.split(".")[0])
-            minor_version = int(version.split(".")[1])
-        except IndexError:
+            major_version, minor_version = version.split(".")
+        except ValueError:
             return Response(
                 {"Query Error: The incorrect version requested"}, status=400
             )
 
-        snap = HistoricalSnapshot.objects.filter(
-            panel__pk=panel_name,
-            major_version=major_version,
-            minor_version=minor_version,
-        ).first()
-
-        if not snap:
-            return Response(
-                {
-                    "Query Error: The version requested for panel:"
-                    + panel_name
-                    + " was not found."
-                }
-            )
-        json = snap.to_api_0()
-        return Response(json)
-
+        queryset = GenePanelSnapshot.objects.get_active(all=True, internal=True, deleted=True)
     else:
         queryset = GenePanelSnapshot.objects.get_active()
 
-        queryset_name_exact = queryset.filter(panel__name=panel_name)
-        if not queryset_name_exact:
-            queryset_name = queryset.filter(panel__name__icontains=panel_name)
-            if not queryset_name:
-                queryset_old_names = queryset_name.filter(
-                    old_panels__icontains=panel_name
-                )
-                if not queryset_old_names:
+    queryset_name_exact = queryset.filter(panel__name=panel_name)
```

Comment:
@Bash-Maistora if the version is specified we only need to check the ids. Names are only for as a search for active panels.